File: CPP/Program/Vehicle.cpp
/*Saya Legi Kuswandi mengerjakan LP4 dalam mata kuliah DPBO
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang 
telah dispesifikasikan. Aamiin.*/

// Menggunakan pragma once agar header file hanya di-include satu kali
#pragma once
// Memasukan library yang digunakan
#include <iostream>
#include <string>
// Deklarasi namespace
using namespace std;
class Vehicle {
private:
    string platnomor;
    string merk;
    string tahun_produksi;
    string warna;
public:
    // Constructor
    Vehicle()
    {

    }

    // Constructor : Overloading
    Vehicle(string platnomor){
        this->platnomor = platnomor;
        merk = "";
        tahun_produksi = "";
        warna = "";
    }
    // Constructor : Overloading
    Vehicle(string platnomor, string merk){
        this->platnomor = platnomor;
        this->merk = merk;
        tahun_produksi = "";
        warna = "";
    }
    // Constructor : Overloading
    Vehicle(string platnomor, string merk, string tahun_produksi){
        this->platnomor = platnomor;
        this->merk = merk;
        this->tahun_produksi = tahun_produksi;
        warna = "";
    }
    // Constructor : Overloading
    Vehicle(string platnomor, string merk, string tahun_produksi, string warna){
        this->platnomor = platnomor;
        this->merk = merk;
        this->tahun_produksi = tahun_produksi;
        this->warna = warna;
    }

    // Method setter untuk set nilai setiap atribut pada kelas Vehicle
    void setPlatnomor(string platnomor){
        this->platnomor = platnomor;
    }

    // Method getter untuk mendapatkan nilai setiap atribut pada kelas Vehicle
    string getPlatnomor(){
        return platnomor;
    }

    // Method setter untuk set nilai setiap atribut pada kelas Vehicle
    void setmerk(string merk){
        this->merk = merk;
    }

    // Method getter untuk mendapatkan nilai setiap atribut pada kelas Vehicle
    string getMerk(){
        return merk;
    }

    // Method setter untuk set nilai setiap atribut pada kelas Vehicle
    void setTahun_produksi(string tahun_produksi){
        this->tahun_produksi = tahun_produksi;
    }

    // Method getter untuk mendapatkan nilai setiap atribut pada kelas Vehicle
    string getTahun_produksi(){
        return tahun_produksi;
    }

    // Method setter untuk set nilai setiap atribut pada kelas Vehicle
    void setwarna(string warna){
        this->warna = warna;
    }

    // Method getter untuk mendapatkan nilai setiap atribut pada kelas Vehicle
    string getwarna(){
        return warna;
    }

    // Destructor
    ~Vehicle()
    {

    }
};

File: Python/Program/Vehicle.py
class Vehicle:
    def __init__(self, platnomor="", merk="", tahun_produksi="", warna=""):
        # Inisialisasi atribut-atribut kendaraan
        self.platnomor = platnomor  # Nomor plat kendaraan
        self.merk = merk  # Merk kendaraan
        self.tahun_produksi = tahun_produksi  # Tahun produksi kendaraan
        self.warna = warna  # Warna kendaraan

    # Setter untuk nomor plat kendaraan
    def set_platnomor(self, platnomor):
        self.platnomor = platnomor

    # Getter untuk nomor plat kendaraan
    def get_platnomor(self):
        return self.platnomor

    # Setter untuk merk kendaraan
    def set_merk(self, merk):
        self.merk = merk

    # Getter untuk merk kendaraan
    def get_merk(self):
        return self.merk

    # Setter untuk tahun produksi kendaraan
    def set_tahun_produksi(self, tahun_produksi):
        self.tahun_produksi = tahun_produksi

    # Getter untuk tahun produksi kendaraan
    def get_tahun_produksi(self):
        return self.tahun_produksi

    # Setter untuk warna kendaraan
    def set_warna(self, warna):
        self.warna = warna

    # Getter untuk warna kendaraan
    def get_warna(self):
        return self.warna

